src/lib.rs

```diff
@@ -39,3 +72,55 @@ mod window_counter;
 // Default implementations of generic interfaces
 pub mod connectors;
 pub mod resolvers;
+
+/// USDT probes for tracing how qorb makes pools and hands out claims.
+#[cfg(feature = "probes")]
+#[usdt::provider(provider = "qorb")]
+mod probes {
+    /// Fires right before attempting to acquire a claim from the pool.
+    fn claim__start(id: &usdt::UniqueId) {}
+
+    /// Fires when a claim is successfully acquired from the pool.
+    fn claim__done(id: &usdt::UniqueId) {}
+
+    /// Fires when we _fail_ to acquire a claim from the pool, with a string
+    /// identifying the reason.
+    fn claim__failed(id: &usdt::UniqueId, reason: &str) {}
+
+    /// Fires right before attempting to make a connection, with the address
+    /// we're connecting to.
+    fn connect__start(id: &usdt::UniqueId, addr: &std::net::SocketAddr) {}
+
+    /// Fires just after successfully making a connection.
+    fn connect__done(id: &usdt::UniqueId) {}
+
+    /// Fires just after failing to make a connectiona, with a string
+    /// identifying the reason.
+    fn connect__failed(id: &usdt::UniqueId, reason: &str) {}
+
+    /// Fires when qorb creates a new handle, before returning it in a call to
+    /// `qorb::Pool::claim()`. This includes an ID for the handle, and the
+    /// address of the backend the handle is connected to.
+    fn handle__claimed(id: usize, addr: &std::net::SocketAddr) {}
+
+    /// Fires when qorb recycles a handle, usually when it is dropped.
+    fn handle__recycled(id: usize) {}
```

Comment:
I might change the name on this one - the name implies that it has been recycled, but this happens when the recycling process *starts*, right?

Maybe `handle__returned` or `handle__unclaimed` or something

Reply:
Yeah, either of those sound good to me. I thought the term "recycled" meant "going back to the pool", which explains the naming `Connection::on_recycle`.

Reply:
I'm mostly beefing with the tense here - recycling may involve doing health checks, or it may involve dropping the handle if we don't need it. Internally, the slot set calls `recycle_connection`, which sets the state of a single slot to `ConnectedRecycling`. Later, a different tokio::task (for the slot) actually does the work of recycling here:

https://github.com/oxidecomputer/qorb/blob/86f6f96294c385000f1454e388fafda9a7555804/src/slot.rs#L628-L633

So, to me, "Recycling" is this whole process of "handle returned to pool, state changed to recycling, and state validated before going back to unclaimed"

Reply:
Got it, thanks for explaining. I've switched to `handle__returned`!